classes/citation/CitationListTokenizerFilter.inc.php

```diff
@@ -36,18 +36,65 @@ function CitationListTokenizerFilter() {
 	 */
 	function &process(&$input) {
 		// The default implementation assumes that raw citations are
-		// separated with line endings.
-		// 1) Remove empty lines and normalize line endings.
-		$input = String::regexp_replace('/[\r\n]+/s', "\n", $input);
-		// 2) Remove trailing/leading line breaks.
+		// separated with a blank line.
+
+		// Normalize line endings.
+		if (strpos($input, "\r") !== FALSE) {
```

Comment:
PHP's single-byte string operations could potentially [scramble UTF-8 data](https://stackoverflow.com/questions/3786003/str-replace-on-multibyte-strings-dangerous) -- especially when working with single bytes e.g. `\r`. Better to use `PKPString` wrappers -- that's also why we use `String::regexp_replace` rather than `preg_replace`.